app/routes.py

```diff
@@ -1,2 +1,205 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request
+from flask import request, Blueprint, make_response
+from flask import jsonify
+from datetime import datetime 
+from sqlalchemy import asc, desc
+import requests
+import json
+import os
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
 
+#This is is a new endpoint.
+@task_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def get_single_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+
+    complete = task.completed_at_helper()  # Helper function to return boolean
+
+    if request.method == "GET":
+        return jsonify({"task": task.json_object()}), 200
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.is_complete = form_data["completed_at"] 
+
+        db.session.commit()
+        return jsonify({"task": task.json_object()}),200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify({"details": f'Task {task.id} "{task.title}" successfully deleted'}), 200
+
+@task_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        
+        request_value = request.args.get("sort") 
+        
+        if request_value == None:
+            tasks = Task.query.all()
+
+        if request_value == "asc":
+            tasks = Task.query.order_by(Task.title.asc()) 
+        
+        
+        if request_value == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        task_response = []
+
+        for task in tasks:
+        
+            task_response.append(task.json_object())
+        
+        return jsonify(task_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        if "completed_at" not in request_body or "description" not in request_body or "title" not in request_body:
+            return jsonify({
+                "details": "Invalid data"
+            }), 400
+
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+        
+        return jsonify({"task": new_task.json_object()}), 201
+
+        db.session.add(new_task)
+        db.session.commit()
```

Comment:
anything after a return statement will never get hit (but I think this was just duplicate code since you have it on lines 78 and 79)

```suggestion

```